### PR TITLE
do not clear old parts at shutdown

### PR DESCRIPTION
**Pull request by CheSema**
Shutdown at tests will be much faster.

Thus is allow to decrease back to the default value max-tries at stop. With was increased up to 3 times here https://github.com/ClickHouse/ClickHouse/pull/43277

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Shutdown will be much faster if do not call clearOldPartsFromFilesystem. Especially this is right for tests with zero-copy due to single thread deletion parts. clearOldPartsFromFilesystem is unnecessary after https://github.com/ClickHouse/ClickHouse/pull/41145

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: docker/test/stress/run.sh [hunk 1]
```diff
@@ -131,13 +131,7 @@ function stop()
     # Preserve the pid, since the server can hung after the PID will be deleted.
     pid="$(cat /var/run/clickhouse-server/clickhouse-server.pid)"
 
-    # --max-tries is supported only since 22.12
-    if dpkg --compare-versions "$(clickhouse local -q 'select version()')" ge "22.12"; then
-        # Increase default waiting timeout for sanitizers and debug builds
-        clickhouse stop --max-tries 180 --do-not-kill && return
```
Comment (tavplubix) [inline docker/test/stress/run.sh:137]: Maybe we should keep increased timeout for BC check (because old versions do not have this improvement)

**Review by tavplubix — APPROVED** [review]

File: docker/test/stress/run.sh (hunk 1 above)
Comment (CheSema) [reply docker/test/stress/run.sh:137]: Btw. I foresee that bc check will need more time to start. It will delete old parts from previous run at start now.

**Review by alesapin — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]
Reverts #8602.

**Comment by CheSema** [comment]
Integration tests asan, failed all cases in test_distributed_queries_stress/test.py::test_stress_distributed.
Test is flaky. https://github.com/ClickHouse/ClickHouse/issues/41776

**Comment by CheSema** [comment]
stress test without check_status.tsv https://github.com/ClickHouse/ClickHouse/issues/41850

**Comment by CheSema** [comment]
02481_async_insert_dedup new test, seems to be flaky.

**Comment by CheSema** [comment]
test_merge_tree_hdfs/test.py::test_attach_detach_partition
Actually test needs to be adjusted. There is wrong way how to wait part deletion.

File: docker/test/stress/run.sh (hunk 1 above)
Comment (tavplubix) [reply docker/test/stress/run.sh:137]: The problem is that old version in BC check does not support `--max-tries` argument:
```

+ clickhouse stop --max-tries 180 --do-not-kill
std::exception. Code: 1001, type: boost::wrapexcept<boost::program_options::unknown_option>, e.what() = unrecognised option '--max-tries' (version 22.11.2.30 (official build))
```
https://s3.amazonaws.com/clickhouse-test-reports/44157/d54f739836fa998f0e32458bcaf3dad545e25526/stress_test__msan_.html

Comment (tavplubix) [reply docker/test/stress/run.sh:137]: And seems like it triggers gdb in BC check all the time and sometimes gdb hangs and it leads to "stress test without check_status.tsv "

**Pull request merged**